sim/druid/moonfire.go

```diff
@@ -0,0 +1,87 @@
+package druid
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+)
+
+func (druid *Druid) registerMoonfireSpell() {
+	// TODO: Genesis increase to Moonfire duration
+	// TODO: Shooting stars proc on periodic damage
+	// TODO: Glyph of Moonfire increase to periodic damage
+	// TODO: Calculate ticks based on haste
+	numTicks := druid.moonfireTicks()
+	//hasMoonfireGlyph := druid.HasPrimeGlyph(proto.DruidPrimeGlyph_GlyphOfMoonfire)
+	//bonusPeriodicDamageMultiplier := core.TernaryFloat64(hasMoonfireGlyph, 0.2, 0)
+
+	druid.Moonfire = druid.RegisterSpell(Humanoid|Moonkin, core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 8921},
+		SpellSchool:    core.SpellSchoolArcane,
+		ProcMask:       core.ProcMaskSpellDamage,
+		ClassSpellMask: DruidSpellMoonfire | DruidArcaneSpells | DruidSpellDoT,
+		Flags:          SpellFlagNaturesGrace | SpellFlagOmenTrigger | core.SpellFlagAPL,
+
+		ManaCost: core.ManaCostOptions{
+			BaseCost:   0.21,
+			Multiplier: 1,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+		},
+
+		DamageMultiplier: 1,
+
+		CritMultiplier:   druid.BalanceCritMultiplier(),
+		ThreatMultiplier: 1,
+		BonusCoefficient: 0.18,
+
+		Dot: core.DotConfig{
+			Aura: core.Aura{
+				Label: "Moonfire",
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+
+				},
```

Comment:
Should not have empty callbacks

Reply:
not sure I even need those right now, so I'll just remove and readd them if necessary.